File: Sources/Utilities/ReadTiff.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
"""
Created on Mon May 11 16:11:32 2015

@author: ta240184
"""
import sys
import hdf5storage
import os
import numpy as np

scriptDirectory = sys.argv[1]
inputDirectory = sys.argv[2]
inputName = sys.argv[3]

sys.path.append(scriptDirectory)
import tifffile as tff

#Read tiff file
inputFileName = inputDirectory+os.sep+inputName

image = tff.imread(inputFileName)
image = image.astype(np.uint16)

#Write temporary mat file
matFile = inputDirectory+os.sep+'temp_'+inputName+'.mat'
print(matFile)
hdf5storage.savemat(matFile,{'temp_image':image})

File: Sources/Utilities/ReadTiff.m
function image=ReadTiff(filename)
%READTIFF Read a 3D image from a tiff file. Uses the python library tifffile.py  
%   input : filename
%   output : image
    
    assert(ischar(filename),'Input argument of ReadTiff must be the name of a tiff file')
    disp('Reading tiff file')
    
    %Get directory to ReadTiff.py
    scriptDirectory = fileparts(mfilename('fullpath'));
    
    pythonScript=fullfile(scriptDirectory,'ReadTiff.py');
    
    %Get absolute name of filename
    curDir = pwd;
    [pathstr, tiffFileName,ext] = fileparts(filename);
    if ~isempty(pathstr)
        cd(pathstr)
    end
    tiffFileDir=pwd; % full (absolute) path
    cd(curDir) % get back to where you were
    
    
    
    %call python script which reads the image in the .tiff file and writes
    %a temporary .mat file
    disp('Calling python function from Matlab')
    commandLine=['python',' ',pythonScript,' ',scriptDirectory,' ',tiffFileDir,' ',tiffFileName,ext];
    disp(commandLine)
    system(commandLine)
    
    
    %load the image in the temporary .mat file
    matFileName=fullfile(tiffFileDir,strcat('temp_',tiffFileName,ext,'.mat'));
    load(matFileName)
    
    image=temp_image;    
    
    delete(matFileName)
end
